discojs/src/privacy.ts

```diff
@@ -12,13 +14,85 @@ async function frobeniusNorm(weights: WeightsContainer): Promise<number> {
   return Math.sqrt(squared[0]);
 }
 
-/** Scramble weights */
-export function addNoise(
+/** Compute the frobenius norm of weight for one layer */
+export async function layerWiseFrobeniusNorm(weight: tf.Tensor): Promise<number>{
+  const squared = await weight.square().sum().data();
+  return Math.sqrt(squared[0]);
+}
+
+/** ALDP-FL implementation */
+// Conditions need to be added for the first three epochs -> get the avg update from all of the available previous updates
+export function getClippingRadius(weightHistory: WeightHistory, defaultRadius:number|undefined = 20, windowSize: number=3, minRadius=1e-12): number | number[]{
```

Comment:
using [default params in js](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) is the same as allowing undef and then setting the variable to the default. so no need to it ourself.

also, the type can be easily infered via the default, no need to set explicitly. and I see that `defaultRadius` & `windowSize` are never set in code so we can move it as constant at the start of the function (I find it a bit more readable and easy to change back if wanted).

```suggestion
export function getClippingRadius(
	weightHistory: WeightHistory,
	defaultRadius = 20,
): number | number[] {
  const WINDOW_SIZE = 3;
  const MINIMUM_RADIUS = 1e-12;
```

you can then avoid the `?? 20` below